app/javascript/pages/components/DataViewer.jsx

```diff
@@ -52,9 +53,13 @@ export default class DataViewer extends React.Component {
         axios.all([tableQuery, headerQuery]).then((response) => {
           const tableResults = response[0];
           const metadata = Object.values(response[1].data)[0];
+          const columns = Object.keys(tableResults.data.rows[0]);
+          const sortedMetadata = metadata.documentation.metadata.eainfo.detailed.attr.map((attribute) => ({ name: attribute.attrlabl, alias: attribute.attalias }))
+            .filter((header) => columns.includes(header.name))
```

Comment:
Didn't catch this before, but you could refactor this to be `.filter((header) => columns.includes(header.name) && header.name !== 'shape)` if you wanted to save a line.